### ISSUE TITLE
Sidecar injection MutatingAdmissionWebhook throws "getsockopt: no route to host" error in API server logs

### ISSUE THREAD

**Issue by jayanthpurushothaman** [istio/old_issues_repo#250]
**Is this a BUG or FEATURE REQUEST?:**
BUG

**Did you review https://istio.io/help/ and existing issues to identify if this is already solved or being worked on?**:
Y

**Bug:**
Y

**What Version of Istio and Kubernetes are you using, where did you get Istio from, Installation details**
```
istioctl version: istio-0.6.0 / istio-0.7.0-pre20180323-09-15
kubectl version: 1.9.3
```

**Is Istio Auth enabled or not ?**
No

**What happened**:
After following the official documentation for Istio 0.6.0 to get the Automatic Sidecar injection enabled on Kubernetes 1.9.4 (including the admission control settings), we tried installing the standard **sleep** application that comes with the samples. However, the automatic sidecar injection doesn't happen even after labelling the namespace. Only manual injection works.

Upon looking at the logs in the API server, the following entries were found:

W0324 15:30:11.360416       1 admission.go:261] Failed calling webhook, failing closed sidecar-injector.istio.io: failed calling admission webhook "sidecar-injector.istio.io": Post https://istio-sidecar-injector.istio-system.svc:443/inject: dial tcp 10.244.223.15:443: getsockopt: no route to host

logging error output: "k8s\x00\n\f\n\x02v1\x12\x06Status\x12\xc4\x03\n\x06\n\x00\x12\x00\x1a\x00\x12\aFailure\x1a\xd0\x01Internal error occurred: failed calling admission webhook \"sidecar-injector.istio.io\": Post https://istio-sidecar-injector.istio-system.svc:443/inject: dial tcp 10.244.223.15:443: getsockopt: no route to host\"\rInternalError*\xcb\x01\n\x00\x12\x00\x1a\x00\"\xbe\x01\n\x00\x12\xb7\x01failed calling admission webhook \"sidecar-injector.istio.io\": Post https://istio-sidecar-injector.istio-system.svc:443/inject: dial tcp 10.244.223.15:443: getsockopt: no route to host\x1a\x00(\x002\x000\xf4\x03\x1a\x00\"\x00"

**What you expected to happen:**
Automatic sidercar injection should work without any issues.

**How to reproduce it:**
Just install any sample Application that comes with istio-0.7.0-pre20180323-09-15 and watch the API server logs.

**Feature Request:**
N

**Comment by yusuoh** [comment]
Did you have a chance to check the status or log for the sidecar injector service pod?

**Comment by freemantle1989** [comment]
Got exactly the same Problem after upgrading from 0.6.0 to 0.7.1

But i have to say that we had Problems in 0.6.0 too.

**Comment by ayj** [comment]
You could try to verify basic network connectivity between your master and the pod running the webhook.

**Comment by ostromart** [comment]
I've picked up webhook from Jason recently. I'm not able to help with older version of webhook but could you please retry with 0.8?

**Comment by ayj** [comment]
The original issue was reported against 0.6.0./0.7.1. Many bugs have been fixed in the 0.8.x and forthcoming 1.0 release. Can those experiencing the above issue please retry with one of those two releases and open new issues accordingly if the problem persists with the latest logs, etc?
